### PR TITLE
Added progress report in algorithms: AddLogDerivative, CalculateDIFC, CalculateZScore, ChopData

### LINKED ISSUE #14291: Add progress report in algorithms: AddLogDerivative, CalculateDIFC, CalculateZScore, ChopData
These algorithm don't seem to report progress.

There are some examples in the usage examples / doc tests that could be used to work on this ticket/issue. 

For an example of algorithm that does report progress status, see SumSpectra or Rebin. I think that those two are good examples.

This may be helpful:

C++ API doxygen page: http://doxygen.mantidproject.org/nightly/de/dd6/classMantid_1_1API_1_1Progress.html
Python: http://www.mantidproject.org/PythonAlgorithm_Progress_Reporting

### PR DESCRIPTION
**Pull request by LamarMoore**
Fixes #14291 

The release notes have not been updated.
## Changes

These algorithms are now reporting progress when executed.
### AddLogDerivative

Run the following script:

http://docs.mantidproject.org/nightly/algorithms/AddLogDerivative-v1.html#id4
### CalculateDIFC

Run the following script:

_ws = LoadEmptyInstrument(Filename="GEM_Definition.xml", OutputWorkspace="ws")_
_ws = CalculateDIFC(ws, OutputWorkspace="ws")_

_#Print the result_
_print "The output workspace has %i spectra" % ws.getNumberHistograms()_
_print "DIFC of pixel %i is %1.f" % (100, ws.readY(100)[0])_

**NB. You will need the UsageData set from http://sourceforge.net/projects/mantid/files/Sample%20Data/UsageData.zip/download**
### CalculateZScore

Run the following script:

http://docs.mantidproject.org/nightly/algorithms/CalculateZscore-v1.html#id5
### ChopData

Run the following script (set NumBanks = 100 and BankPixelWidth = 20):

http://docs.mantidproject.org/nightly/algorithms/ChopData-v1.html#id5

### CODE REVIEW

**Comment by KarlPalmen** [comment]
The algorithm AddLogDerivative seems to be too fast to need progress reporting. If this is not true, please provide an example test that would demonstrate it.

**Comment by LamarMoore** [comment]
You are correct. The progress reporting is unnecessary.

**Comment by LamarMoore** [comment]
I spoke to Martyn and said that although the progress reporting is unnecessary there is actually no harm leaving it there.

**Comment by KarlPalmen** [comment]
OK. I'll pass it if it work without error or crash and the green progress bar flashes on.

**Comment by LamarMoore** [comment]
Ok thanks

File: Framework/Algorithms/inc/MantidAlgorithms/CalculateDIFC.h [hunk 1]
```diff
@@ -4,6 +4,17 @@
...
+#include "MantidGeometry/IDetector.h"
+
+using Mantid::API::MatrixWorkspace;
+using Mantid::API::WorkspaceProperty;
+using Mantid::DataObjects::OffsetsWorkspace;
+using Mantid::DataObjects::OffsetsWorkspace_sptr;
+using Mantid::DataObjects::SpecialWorkspace2D;
+using Mantid::DataObjects::SpecialWorkspace2D_sptr;
+using Mantid::Geometry::Instrument_const_sptr;
+using Mantid::Kernel::Direction;
```
Comment (peterfpeterson) [inline Framework/Algorithms/inc/MantidAlgorithms/CalculateDIFC.h]: `using` in headers are a bad idea. They should only appear in source files.

**Pull request merged**